sdk/python/feast/feature_server.py

```diff
@@ -145,15 +146,22 @@ def get_online_features(body=Depends(get_body)):
                     resource=od_feature_view, actions=[AuthzedAction.READ_ONLINE]
                 )
 
-        response_proto = store.get_online_features(
+        read_params = dict(
             features=features,
             entity_rows=entity_rows,
             full_feature_names=full_feature_names,
-        ).proto
+        )
+
+        if store._get_provider().async_supported.online.read:
+            response = await store.get_online_features_async(**read_params)
+        else:
+            response = await run_in_threadpool(
+                lambda: store.get_online_features(**read_params)
+            )
```

Comment:
use async feature lookups if it's supported by the online store, otherwise run the sync method in the threadpool